tmt/steps/discover/__init__.py

```diff
@@ -372,7 +373,20 @@ def requires(self) -> list['tmt.base.Dependency']:
 
         :returns: a list of requirements, with duplicaties removed.
         """
-        return flatten((test.require for test in self.tests(enabled=True)), unique=True)
+
+        tests = self.tests(enabled=True)
+
+        return flatten(
+            itertools.chain(
+                (test.require for test in tests),
+                # cast: mypy believes we cannot merge `list[Depencency]` and
+                # `list[DependencySimple]`. I think we can, let's see what pyright
+                # will tell us once enabled.
+                (cast(list['tmt.base.Dependency'],
+                      test.test_framework.get_requirements(test, self._logger))
+                 for test in tests)
```

Comment:
Is this really necessary? Aren't the framework requirement already [covered by the changes](https://github.com/teemtee/tmt/pull/2517/files#diff-2f45e2684afbb7caf4a24add9251b6ca1c4d678c04326cc830d7597d93d29f65R277-R280) in `steps/prepare/__init__.py`?

Reply:
Yeah, it's probably not necessary, especially with next changes in https://github.com/teemtee/tmt/pull/2622/files#diff-5f59acdec9a1fdc81c556e6ca1c7c872a25d72f1be0259a8f727b9dce4a45a44L366 where `requires()` gets dropped completely. From this moment, it's not called at all, yet I sort of tried to make it return sane data. It returns "plugin and test requirements", so I included the framework requirements as well. In the next PR, #2622, I realize this method becomes pointless, and gets removed, but in the meantime, it will work and return "requirements". It's quite likely nobody starts using it before it gets dropped, but who knows... It's definitely not needed now, but that becomes clearer with "essential requirements".